Improve lock developer documentation (#3226)
home-assistant/developers.home-assistant: docs/core/entity/lock.md

| Name | Type | Default | Description
| ---- | ---- | ------- | -----------
| changed_by | string | None | Describes what the last change was triggered by.
| code_format | string | None | Regex used to validate the code supplied to the lock, unlock, and open actions, or None if no code is required.

> | is_locked | bool | None | Whether the lock is secured. Used to determine `state`.

Reviewer: don't really like the "secured" here. This reads like if the lock itself is secured. Maybe something regarding the door not being openable and the deadbolt being extracted?

> | is_locked | bool | None | Whether the lock is secured. Used to determine `state`.

Author: ```suggestion
| is_locked | bool | None | Whether the lock is locked. Used to determine `state`.
```

| is_locking | bool | None | Whether the lock is in the process of being locked. Used to determine `state`.
| is_unlocking | bool | None | Whether the lock is in the process of being unlocked. Used to determine `state`.

> | is_open | bool | None | Whether the lock is not secured and has released its latch. Only relevant for locks that support `LockEntityFeature.OPEN`. Used to determine `state`.

Reviewer: don't like the "secured" here either

> | is_open | bool | None | Whether the lock is not secured and has released its latch. Only relevant for locks that support `LockEntityFeature.OPEN`. Used to determine `state`.

Author: ```suggestion
| is_open | bool | None | Whether the lock is unlocked and has released its latch. Only relevant for locks that support `LockEntityFeature.OPEN`. Used to determine `state`.
```

| is_opening | bool | None | Whether the lock is in the process of releasing its latch. Used to determine `state`.
| is_jammed | bool | None | Whether the lock tried to move but got stuck before it finished. Used to determine `state`.

### States

The state is derived from the properties above and is one of the following `LockState` enum members:

| Value       | Description                                                        |
|-------------|--------------------------------------------------------------------|
| `LOCKED`    | The lock is secured.                                               |
| `LOCKING`   | The lock is in the process of being locked.                        |

> | `UNLOCKED`  | The lock is not secured.                                           |

Reviewer: ```suggestion
| `LOCKED`    | The lock is locked.                                                                 |
| `LOCKING`   | The lock is in the process of being locked.                        |
| `UNLOCKED`  | The lock is not locked.                                                       |
```
you kept "locked" elsewhere, should we keep it here too?
